core/src/test/java/org/elasticsearch/index/engine/InternalEngineTests.java

```diff
@@ -135,6 +134,7 @@
 import static java.util.Collections.emptyMap;
 import static org.elasticsearch.index.engine.Engine.Operation.Origin.PRIMARY;
 import static org.elasticsearch.index.engine.Engine.Operation.Origin.REPLICA;
+import static org.hamcrest.CoreMatchers.instanceOf;
```

Comment:
commenting here because it's the first line - can we have a test that mock lucene's addDocuments to throw an non-tragic exception and see that is caught as a document one?